test/models/commit_status_test.rb

```diff
@@ -17,156 +17,302 @@ def self.deploying_a_previous_release
     end
   end
 
-  def success!
-    stub_github_api(url, statuses: [{foo: "bar", updated_at: 1.day.ago}], state: "success")
-  end
-
   def failure!
     stub_github_api(url, nil, 404)
   end
 
-  def status(stage_param: stage, reference_param: reference)
-    CommitStatus.new(stage_param.project, reference_param, stage: stage_param)
+  def status(stage_param: stage, reference_param: reference, new: false)
+    if new
+      @status = CommitStatus.new(stage_param.project, reference_param, stage: stage_param)
+    else
+      @status ||= CommitStatus.new(stage_param.project, reference_param, stage: stage_param)
+    end
```

Comment:
methods with arguments should not cache without the arguments ... change the tests to have local variables or do a `let(:status)` that calls `build_status` with default values or so